addons/voice/org.openhab.voice.gtts/src/main/java/org/openhab/voice/gtts/internal/GoogleCloudAPI.java

```diff
@@ -0,0 +1,350 @@
+     *
+     * @return Set of locales
+     */
+    Set<Locale> getSupportedLocales() {
+        return Collections.unmodifiableSet(voices.keySet());
+    }
+
+    /**
+     * Supported voices for locale.
+     *
+     * @param l Locale
+     * @return Set of voices
+     */
+    Set<GoogleTTSVoice> getVoicesForLocale(Locale l) {
+        Set<GoogleTTSVoice> set = voices.get(l);
+        return set != null ? Collections.unmodifiableSet(set) : Collections.EMPTY_SET;
+    }
+
+    /**
+     * Google API call to load locales and voices.
+     */
+    private void initVoices() {
+        if (googleClient != null) {
+            voices.clear();
+            ListVoicesResponse resp = googleClient.listVoices(""); //without language code to get all the voices
+            for (Voice v : resp.getVoicesList()) {
+                for (int i = 0; i < v.getLanguageCodesCount(); i++) {
+                    String languageCode = v.getLanguageCodes(i);
+                    Locale l = Locale.forLanguageTag(languageCode);
+                    Set<GoogleTTSVoice> localVoices;
+                    if (!voices.containsKey(l)) {
+                        localVoices = new HashSet<>();
+                        voices.put(l, localVoices);
+                    } else {
+                        localVoices = voices.get(l);
+                    }
+                    localVoices.add(new GoogleTTSVoice(l, v.getName(), v.getSsmlGenderValue()));
+                }
+            }
+        } else {
+            logger.error("Google client is not initialized!");
+        }
+    }
+
+    /**
+     * Converts openHAB audio format to Google parameters.
+     *
+     * @param codec Requested codec
+     * @return String array of Google audio format and the file extension to use.
+     */
+    private String[] getFormatForCodec(String codec) {
+        switch (codec) {
+            case AudioFormat.CODEC_MP3:
+                return new String[]{AudioEncoding.MP3.toString(), "mp3"};
+            case AudioFormat.CODEC_PCM_SIGNED:
+                return new String[]{AudioEncoding.LINEAR16.toString(), "wav"};
+            default:
+                throw new IllegalArgumentException("Audio format " + codec + " is not yet supported");
+        }
+    }
+
+    File synthesizeSpeech(String text, GoogleTTSVoice voice, String codec) {
+        String[] format = getFormatForCodec(codec);
+        String fileNameInCache = getUniqueFilenameForText(text, voice.getLocale());
+        // check if in cache
+        File audioFileInCache = new File(cacheFolder, fileNameInCache + "." + format[1]);
+        if (audioFileInCache.exists()) {
+            logger.debug("Audio file {} was found in cache.", audioFileInCache.getName());
+            return audioFileInCache;
+        }
+
+        // if not in cache, get audio data and put to cache
+        try (InputStream is = synthesizeSpeechByGoogle(text, voice, format[0]);
+             FileOutputStream fos = new FileOutputStream(audioFileInCache)) {
```

Comment:
Please fix the indentation.

Reply:
I thing this is OK. This is the default indentation of IntelliJ for a try-with-resource block.

Reply:
It's indented with one extra space. :mag::nerd_face:

See the following next lines:

https://github.com/openhab/openhab2-addons/blob/06d504fd24729dbeea51859a7b55d651ab997458/addons/voice/org.openhab.voice.gtts/src/main/java/org/openhab/voice/gtts/internal/GoogleCloudAPI.java#L212-L217

Reply:
I still don't agree... :) 
This is the try block head:
try (InputStream is = synthesizeSpeechByGoogle(text, voice, format[0]); 
      FileOutputStream fos = new FileOutputStream(audioFileInCache)) {
The first line in the block is 
logger.debug("Caching audio file {}", audioFileInCache.getName());

Reply:
Yes you're right and I got confused with just one space of indentation. I checked in Eclipse what the formatter does. It indents these with 4 spaces instead of 1 which makes this less confusing. ;-)